### PR TITLE
Don't bind /etc/host and /etc/resolv.conf if network is not present

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
If there's no network present in the container, don't bind
the /etc/hosts file as any changes to from the caller will
be lost when run exits.  Ditto the /etc/resolv.conf file.

Addresses: #2478

Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

### CODE REVIEW

File: tests/run.bats [hunk 1]
```diff
@@ -422,3 +422,40 @@ function configure_and_check_user() {
...
+        run_buildah from --quiet --pull=false --signature-policy ${TESTSDIR}/policy.json alpine
+        cid=$output
+        run_buildah run --isolation=chroot --network=none $cid sh -c 'echo "110.110.110.0 fake_host" >> /etc/hosts; cat /etc/hosts'
+	expect_output "110.110.110.0 fake_host"
+        m=$(buildah mount $cid)
+	run cat $m/etc/hosts
+	[ "$status" -eq 0 ]
+	expect_output "110.110.110.0 fake_host"
+	run_buildah rm -a
+}
```
Comment (TomSweeneyRedHat) [inline tests/run.bats:461]: Ugh,

tab/spaces issues, will fix.

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@nalind @mheon @vrothberg @QiWang19 @ashley-cui PTAL

**Comment by QiWang19** [comment]
LGTM

**Comment by rhatdan** [comment]
Looks like either we have a bug or a test needs to be adjusted.

**Comment by TomSweeneyRedHat** [comment]
network error in an unrelated test, bors be kind this time:
bors retry

File: tests/from.bats [hunk 2]
```diff
@@ -255,7 +255,7 @@ load helpers
   _prefetch alpine
   run_buildah from --quiet --add-host=localhost:127.0.0.1 --pull --signature-policy ${TESTSDIR}/policy.json alpine
   cid=$output
-  run_buildah run $cid -- cat /etc/hosts
```
Comment (TomSweeneyRedHat) [inline tests/from.bats:258]: @nalind, does this look hip to you?  This is my one worry spot with this change.  Debian and Alpine act differently.  If you take the "stock" alpine image and run with current upstream Buildah, the run command without the `--net-option` would retain the following in the containers /etc/host file in the containers storage on the host:
```
127.0.0.1	localhost localhost.localdomain
::1		localhost localhost.localdomain
```

And if you investigate the container when it's running, you first see the "# Buildah Generated" lines, then these two lines.  However, the Buildah Generated lines are not retained on disk for the container, only these two lines.

Debian however, has nothing except the "# Buildah Generated" lines when it too is running and then nothing in the /etc/host in the container storage on the host.

With these changes, that file in container storage is empty for both distros.  In both this variant of Buildah and current upstream, the host specified in --add-host is not retained when the run command completes.

Comment (rhatdan) [reply tests/from.bats:258]: Do a buildah mount and see what the contents of the /etc/hosts file are. these hosts files saved to the image, are from the underlying image, which is why you are seeing the difference I believe.  
I don't know why in a running container the localhost data provided by Buildah would be different.  
I don't think we ever save the contents of /etc/hosts or /etc/resolv.conf into a container image.

File: run_linux.go [hunk 3]
```diff
@@ -192,7 +192,10 @@ func (b *Builder) Run(command []string, options RunOptions) error {
 		if err != nil {
 			return err
 		}
-		bindFiles["/etc/hosts"] = hostFile
+		// Only bind /etc/hosts if there's a network
+		if configureNetwork {
```
Comment (rhatdan) [inline run_linux.go:196]: Should we do the same thing for resolv.conf?

File: tests/from.bats (hunk 2 above)
Comment (TomSweeneyRedHat) [reply tests/from.bats:258]: If I do `buildah run {alpine||debian}-working-container cat /etc/hosts`  on both distros I see:

```
# Generated by Buildah
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
127.0.0.1   0f8200b7e866
::1         0f8200b7e866
```
If I mount them, then two different but similar lines in the alpine mount, and none in Debian.

```
# ma=$(buildah mount alpine-working-container)
# mb=$(buildah mount debian-working-container)

# cat $ma/etc/hosts
127.0.0.1	localhost localhost.localdomain
::1		localhost localhost.localdomain

# cat $mb/etc/hosts

```
So it looks like we're getting something from alpine directly and it's not part of the generated one that we make when running.  I'm not quite sure how/where that's coming from.

The main focus of this fix is just to retain whatever the user adds to /etc/hosts if `--net=none`

File: run_linux.go (hunk 3 above)
Comment (TomSweeneyRedHat) [reply run_linux.go:196]: On the fence.  Do you or others want to tip me one way or the other @rhatdan ?  I'm leaning towards yes, but not enough to tip over yet.

**Comment by TomSweeneyRedHat** [comment]
I've repushed, the from.bats test was apparently still failing, but it works in my sandbox.  I'm not sure the push will fix anything, but we'll see.

File: run_linux.go (hunk 3 above)
Comment (mrinaldhillon) [reply run_linux.go:196]: IMHO /etc/hosts /etc/hostname /etc/resolve.conf should not be managed with net=none

Comment (rhatdan) [reply run_linux.go:196]: I agree.

Comment (TomSweeneyRedHat) [reply run_linux.go:196]: Sold!

I'll add `/etc/resolv.conf` too to the PR and will push a new variant out later today or Monday.

**Comment by TomSweeneyRedHat** [comment]
Turns out the `configureNetwork` changes holds when doing `buildah run`, but does not hold when there's a `RUN` statement in a Containerfile.  I played with it a bit last night, and think I've a solution for both, more testing today.

**Comment by TomSweeneyRedHat** [comment]
All happy green test buttons.  This is ready to go in with appropriate reviews.  I would like a @nalind head nod if he's got a minute or two.

**Comment by rhatdan** [comment]
bors r+
@TomSweeneyRedHat could you look to see if we have the same issue with Podman?

**Comment by TomSweeneyRedHat** [comment]
Created an investigative issue on Podman so I won't forget.   https://github.com/containers/podman/issues/7206

**Pull request merged**